pkg/apis/pipeline/v1/task_validation.go

```diff
@@ -268,6 +268,7 @@ func validateStep(ctx context.Context, s Step, names sets.String) (errs *apis.Fi
 		if !config.FromContextOrDefaults(ctx).FeatureFlags.EnableStepActions {
 			return apis.ErrGeneric("feature flag %s should be set to true to reference StepActions in Steps.", config.EnableStepActions)
 		}
+		errs = s.Ref.Validate(ctx)
```

Comment:
Trying to understand if there're any specific reason why don't we add the validation of the field also in this file?

Reply:
Sorry I don't get the question

Reply:
Oh did you mean move the validation function to this file?

Reply:
Yep, I saw pipelineRef/taskRef validation has their own file probably because they are a struct. To me, it might be easier to understand if we move the `validateStep` to this Container_validation.go or move this `validate` func for Ref to the task_validation.

But either way, not to be blocking. They could both be in another PR or a TODO.

Reply:
I think it is better to have a separate ref.go file, ref_validation.go etc. We can always clean up them later, 😛